tests/test_max_weight_matching.py

```diff
@@ -0,0 +1,552 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+# These tests are adapated from the networkx test cases:
+# https://github.com/networkx/networkx/blob/3351206a3ce5b3a39bb2fc451e93ef545b96c95b/networkx/algorithms/tests/test_matching.py
+
+from itertools import combinations
+import random
+import unittest
+
+import retworkx
+import networkx
+
+
+def match_dict_to_set(match):
+    return {(u, v) for (u, v) in set(map(frozenset, match.items()))}
+
+
+def is_matching(matching):
+    return all(
+        len(set(e1) & set(e2)) == 0 for e1, e2 in combinations(matching, 2))
+
+
+def is_maximal_matching(graph, rx_matches):
```

Comment:
`is_matching` and `is_maximal_matching` might be useful if they are included as part of retworkx. What do you think?

Reply:
Yeah I agree, it'll be good thing to add as a retworkx function in a follow up PR (this one is already big enough :) ). I've opened an issue on this here: #255.